obstacleAvoidanceWIthSmartCar/src/obstacleAvoidanceWithSmartCar.ino

```diff
@@ -0,0 +1,119 @@
+#include <Smartcar.h>
+
+// Configuration of left and right odometers, from SmartCar Library example.
+DirectionlessOdometer leftOdometer(50), rightOdometer(60);
+const int LEFT_ODOMETER_PIN = 2;
+const int RIGHT_ODOMETER_PIN = 3;
+
+// Config of gyroscopePrintAngle
+//GYROSCOPE_OFFSET is set to 9 because of calibration purposes
+const int GYROSCOPE_OFFSET = 9;
+GY50 gyroscope(GYROSCOPE_OFFSET);
+
+// Control configuration
+BrushedMotor leftMotor(8, 10, 9);
+BrushedMotor rightMotor(12, 13, 11);
+
+// DifferentialControl: controling a vehicle by
+// applying different speed on each side, like a tank.
+DifferentialControl control(leftMotor, rightMotor);
+SmartCar car(control, gyroscope, leftOdometer, rightOdometer);
+
+//variables for the obstacle avoidance
+const int TRIGGER_PIN = 6; //D6
+const int ECHO_PIN = 5; //D5
+const unsigned int MAX_DISTANCE = 30;
+SR04 front(TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+int distance;
+
+// Setup code runs once
+void setup() {
+  Serial.begin(9600);
+  odometerSetUp();
+
+  // Cruise Control controls the car speed in meters/second
+  //using default PID values
+  // car.enableCruiseControl();
+}
+
+// Loop code runs repeatedly
+void loop() {
+  measureDistance();
+  setCarMoveForward();
+
+  // && distance > 0 because of Arduino sensor bug
+  if(distance < MAX_DISTANCE && distance > 0 ){
+    stopCar();
+    changeRandomDirection();
+  };
+}
+
+// Randomly chooses left or right and turn continuously until way is free
+void changeRandomDirection() {
+  float randomNumber = generateRandomNumber();
+  while (distance < MAX_DISTANCE && distance > 0 ) {
+    if (randomNumber > 4) {
+      changeDirectionRight();
+    }
+    else {
+      changeDirectionLeft();
+    };
+    measureDistance();
+  }
+}
+
+long generateRandomNumber() {
+  return random(0, 10);
+}
+
+void changeDirectionRight() {
+  //Arguments(leftMotor speed capacity, rightMotor speed capacity)
+  car.overrideMotorSpeed(50, -50);
+  //delay so the car has enough time to turn
+  delay(150);
```

Comment:
Why are you using the `delay` here? How does it help?

In any case, this is meant to be a blocking function by design, so my comments against the use of blocking delays are not really valid. That being said, I still think it's not necessary.

Reply:
Thank you for your input Dimitris. We'll merge the code as it is and create a new issue to improve the methods which are using the delay() function. We'll get back to you about this to talk about the new code.

One problem with using the delay() that we have detected would be if the user hit the "stop alarm" button while the car is on delay(150). Although the occurrence rate of this would be small, it would be frustrating for the user experience.

Reply:
> One problem with using the delay() that we have detected would be if the user hit the "stop alarm" button while the car is on delay(150).

This is actually not a problem because there's much a bigger one if this is your intended behavior. Specifically, if the user wants to "stop the alarm" then the car will keep moving until "the way is free" irrespective of the user input.

Reply:
Again, good input Dimitris. Thank you very much!
We'll keep this in mind and tackle it in the next two sprints.